src/core/repositories/AdminRepository.ts

```diff
@@ -0,0 +1,41 @@
+import { FilterQuery, UpdateQuery } from 'mongoose';
+import { IAdmin } from '../interfaces/auth';
+import { Admin, IAdminModel } from '../models/admin';
+
+export class AdminRepository {
+  async createAdmin(payload: Omit<IAdmin, '_id'>): Promise<IAdmin> {
+    const admin = await Admin.create(payload);
+    const obj = admin.toObject();
+    obj._id = admin._id.toString();
+    return obj;
```

Comment:
Is there a reason why we are not returning admin directly:
```suggestion
    return admin;
```

Reply:
Done